hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/datanodes/datanodes.tsx

```diff
@@ -436,8 +453,38 @@ export class Datanodes extends React.Component<Record<string, object>, IDatanode
     console.log(current, pageSize);
   };
 
+  onSelectChange = (newSelectedRowKeys:any) => {
+    this.setState({
+      selectedRowKeys: newSelectedRowKeys
+    });
+  };
+
+  onDisable = (record:any) => {
+    //Enable Record for Remove whos state is Dead or operation state Decommissioned or In Maintenance
+    if (record.state !== 'DEAD') {
+      return record.opState === 'IN_SERVICE' || record.opState === 'ENTERING_MAINTENANCE' || record.opState === 'DECOMMISSIONING';
+    }
```

Comment:
This isn't doing what the comments says. Based on the comment you'd like to enable the datanode deletion if either its state is dead or its operational state is decommissioned or in maintenance. 
The code checks if it's dead and it only later checks based on the operational state. E.g. if a datanode is not dead, but in maintenance it won't let it to be removed, but based on the comment and the PR description it should act differently. 
I'm not 100% sure what states can a decommissioned or in maintenance operational stated datanode can be in, but I'd go with a steady solution and check for all three conditions on the same level.

Reply:
@dombizita 

Corrected Comments in latest commits.